modules/services/pass-secret-service.nix

```
@@ -2,31 +2,47 @@
 
 with lib;
 
-let serviceCfg = config.services.pass-secret-service;
+let cfg = config.services.pass-secret-service;
 in {
   meta.maintainers = [ maintainers.cab404 ];
+
   options.services.pass-secret-service = {
     enable = mkEnableOption "Pass libsecret service";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pass-secret-service;
+      defaultText = literalExpression "pkgs.pass-secret-service";
+      description = "Package providing <command>pass_secret_service</command>.";
+    };
+
+    storePath = mkOption {
+      type = types.str;
+      default = "$PASSWORD_STORE_DIR";
+      defaultText = "$PASSWORD_STORE_DIR";
+      example = "/home/user/.local/share/password-store";
+      description = "Absolute path to password store.";
+    };
   };
-  config = mkIf serviceCfg.enable {
+
+  config = mkIf cfg.enable {
     assertions = [
       (hm.assertions.assertPlatform "services.pass-secret-service" pkgs
         platforms.linux)
-
-      {
-        assertion = config.programs.password-store.enable;
-        message = "The 'services.pass-secret-service' module requires"
-          + " 'programs.password-store.enable = true'.";
-      }
     ];
 
     systemd.user.services.pass-secret-service = {
-      Unit = { Description = "Pass libsecret service"; };
+      Unit = {
+        Description = "Pass libsecret service";
+        PartOf = [ "default.target" ];
+      };
+
       Service = {
         # pass-secret-service doesn't use environment variables for some reason.
         ExecStart =
-          "${pkgs.pass-secret-service}/bin/pass_secret_service --path ${config.programs.password-store.settings.PASSWORD_STORE_DIR}";
+          "${cfg.package}/bin/pass_secret_service --path ${cfg.storePath}";
```

Can take the opportunity to do

```suggestion
          "${cfg.package}/bin/pass_secret_service --path ${escapeShellArg cfg.storePath}";
```

so that, e.g., spaces in the path won't cause problems.

---

Added in 80e43c10